remotes/handlers.go

```diff
@@ -199,13 +200,20 @@ func push(ctx context.Context, provider content.Provider, pusher Pusher, desc oc
 	return content.Copy(ctx, cw, rd, desc.Size, desc.Digest)
 }
 
+// ContentInfoProvider is a subset of the content.Manager info used by
+// the push operation to get content labels.
+type ContentInfoProvider interface {
+	Info(ctx context.Context, dgst digest.Digest) (content.Info, error)
+}
+
```

Comment:
I see this same signature now being defined as part of a couple of interfaces (also some not in this repo)

https://github.com/containerd/containerd/blob/b0a7f08aa7490a3d43787b9f2b444e4b20b96f8c/pkg/display/manifest_printer.go#L62-L66
https://github.com/containerd/containerd/blob/b0a7f08aa7490a3d43787b9f2b444e4b20b96f8c/content/content.go#L114-L119


While it's good to define the interface on the receiver-side, I"m wondering if we should add it as `content.InfoProvider` in the `content` package, so that we can compose those by embedding it (as it already defines various reusable interfaces);

Those would then become something like;

```go
type ContentReader interface {
	content.Provider
	content.InfoProvider
}
```

And

```go
type InfoProvider interface {
	// Info will return metadata about content available in the content store.
	//
	// If the content is not present, ErrNotFound will be returned.
	Info(ctx context.Context, dgst digest.Digest) (Info, error)
}

// Manager provides methods for inspecting, listing and removing content.
type Manager interface {
	InfoProvider

	// Update updates mutable information related to content.
	// If one or more fieldpaths are provided, only those
	// fields will be updated.
	// Mutable fields:
	//  labels.*
	Update(ctx context.Context, info Info, fieldpaths ...string) (Info, error
	// ...
}
```